geode-redis/src/integrationTest/java/org/apache/geode/redis/general/ExistsIntegrationTest.java

```diff
@@ -0,0 +1,348 @@
+    String hyperLogLogKey = "crawled:127.0.0.2";
+    String hyperLogLogValue = "www.insideTheHouse.com";
+
+    jedis.pfadd(hyperLogLogKey, hyperLogLogValue);
+    jedis.del(hyperLogLogKey);
+
+    assertThat(jedis.exists(toArray(hyperLogLogKey))).isEqualTo(0L);
+  }
+
+  @Test
+  public void shouldReturn1_givenListExists() {
+    String listKey = "listKey";
+    String listValue = "listValue";
+
+    jedis.lpush(listKey, listValue);
+
+    assertThat(jedis.exists(toArray(listKey))).isEqualTo(1L);
+  }
+
+  @Test
+  public void shouldReturn0_givenListDoesNotExist() {
+    String listKey = "listKey";
+    String listValue = "listValue";
+
+    jedis.lpush(listKey, listValue);
+    jedis.del(listKey);
+
+    assertThat(jedis.exists(toArray(listKey))).isEqualTo(0L);
+  }
+
+  @Test
+  public void shouldReturn1_givenBitMapExists() {
+    String bitMapKey = "bitMapKey";
+    long offset = 1L;
+    String bitMapValue = "0";
+
+    jedis.setbit(bitMapKey, offset, bitMapValue);
+
+    assertThat(jedis.exists(toArray(bitMapKey))).isEqualTo(1L);
+  }
+
+  @Test
+  public void shouldReturn0_givenBitMapDoesNotExist() {
+    String bitMapKey = "bitMapKey";
+    long offset = 1L;
+    String bitMapValue = "0";
+
+    jedis.setbit(bitMapKey, offset, bitMapValue);
+    jedis.del(bitMapKey);
+
+    assertThat(jedis.exists(toArray(bitMapKey))).isEqualTo(0L);
+  }
+
+  @Test
+  public void shouldReturnTotalNumber_givenMultipleKeys() {
+    String key1 = "key1";
+    String key2 = "key2";
+
+    jedis.set(key1, "value1");
+    jedis.set(key2, "value2");
+
+    assertThat(jedis.exists(toArray(key1, "doesNotExist1", key2, "doesNotExist2"))).isEqualTo(2L);
+  }
+
+  @Test
+  public void shouldCorrectlyVerifyKeysExistConcurrently() throws InterruptedException {
+    int iterationCount = 5000;
+    setKeys(jedis, iterationCount);
+
+    AtomicLong existsCount = new AtomicLong(0);
+
+    Thread thread1 =
+        new LoopingThread((i) -> existsCount.addAndGet(jedis.exists(toArray("key" + i))),
+            iterationCount);
+    Thread thread2 =
+        new LoopingThread((i) -> existsCount.addAndGet(jedis2.exists(toArray("key" + i))),
+            iterationCount);
+
+    thread1.start();
+    thread2.start();
+    thread1.join();
+    thread2.join();
+
+    assertThat(existsCount.get()).isEqualTo(2 * iterationCount);
+  }
+
+  @Test
+  public void shouldCorrectlyVerifyKeyExistsConcurrentlyWhileDeletingAndSettingKey()
+      throws InterruptedException {
+    int iterationCount = 5000;
+
+    Thread loopingThread1 = new LoopingThread((i) -> jedis.set("key", "value"), iterationCount);
+    Thread loopingThread2 = new LoopingThread((i) -> jedis2.exists(toArray("key")), iterationCount);
+    Thread loopingThread3 = new LoopingThread((i) -> jedis3.del("key"), iterationCount);
+
+    loopingThread1.start();
+    loopingThread2.start();
+    loopingThread3.start();
+    loopingThread1.join();
+    loopingThread2.join();
+    loopingThread3.join();
+  }
```

Comment:
Nothing to assert? Are we just happy that none of the threads throws an exception? If so the method name should communicate this.

Reply:
updated, the name. We just wanted to make sure there were no Concurrent Modification Exceptions on any underlying data structures.  Not sure what else to test. thx